### ISSUE TITLE
Empty array in composite type is wrongly treated as NULL if field's type is Option[List[A]]

### ISSUE THREAD

**Issue by danishin** [tminglei/slick-pg#490]
Empty array in composite type is wrongly treated as `NULL` if field's type is `Option[List[A]]`

For example,
```
case class FooComposite(optional_list1: Option[List[Int]]) extends Struct

FooComposite(None) // correctly returned as `()`
FooComposite(Some(List.empty)) // incorrectly returned as `()`
FooComposite(Some(List(1))) // correctly returned as `({})`
```

If I change type to `List[A]` then it works fine (return `FooComposite(Some(List.empty))` as `({})`).

**Comment by tminglei** [comment]
@danishin I checked it with latest codes. It seems ok.
Maybe it was fixed by #488

**Comment by danishin** [comment]
@tminglei That's great news. Thanks for the follow up!
Always appreciate your time and efforts to maintain this repo. It's been helping our company lots!
